### PR TITLE
Update Base Class of Metric

### PR DESCRIPTION
**Pull request by MooooCat**
## Description

Updated several files in the `sdgx/metric/` directory

## Motivation and Context

Update the metric base class used to evaluate the quality of synthetic data

## How has this been tested?

After the specific metric is implemented, that metric will be integrated into the `test/` directory and tested using pytest.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: sdgx/metrics/column/base.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+
+class columnMetric(object):
```
Comment (Wh1isper) [inline sdgx/metrics/column/base.py:4]:
```suggestion
class ColumnMetric(object):
```

File: sdgx/metrics/single_table/base.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class singleTableMetric:
```
Comment (Wh1isper) [inline sdgx/metrics/single_table/base.py:1]:
```suggestion
class SingleTableMetric:
```

File: sdgx/metrics/single_table/base.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        pass
+
+    @classmethod
+    def check_input(real_data, synthetic_data):
+        # should be pd.DataFrame
+
+        raise NotImplementedError()
+
+    # not a class method
+    def calculate(real_data, synthetic_data):
```
Comment (Wh1isper) [inline sdgx/metrics/single_table/base.py:18]:
```suggestion
    def calculate(self, real_data, synthetic_data):
```

**Review by Wh1isper — CHANGES_REQUESTED** [review]

**Comment by Wh1isper** [comment]
Metric isn't the focus at the moment, I'm in favour of us being able to put a demo on it first, but it might not look like this.

**Comment by Wh1isper** [comment]
When we consider `metrics` as a standalone module, what we expect is basically to input the generated table with the original table. Then `calculate` something like a score or an analysis report. 

But considering that the generated and raw data are **not necessarily** csv files and pandas can't handle very large amounts of data, then we will get a **restricted metrics** class(mostly by performance). Maybe we should separate it to model and data interface.

**Comment by MooooCat** [comment]
> Metric isn't the focus at the moment, I'm in favour of us being able to put a demo on it first, but it might not look like this.

From an algorithmic perspective, metric can become a standalone module (so as to models). We now need to complete the entire process through adding the metric module.

Currently @sjh120  is implementing Jensen–Shannon Divergence(JSD Feature) and adding it to the demo of single-table data generation example, we can suggest him to merge JSD Feature and new demo into this branch.

**Comment by Wh1isper** [comment]
> and adding it to the demo of single-table data generation example

Good! It would make sense if this PR could contain working demos.

**Comment by MooooCat** [comment]
> But considering that the generated and raw data are **not necessarily** csv files and pandas can't handle very large amounts of data, then we will get a **restricted metrics** class(mostly by performance). Maybe we should separate it to model and data interface.


This is indeed an important factor and has been ignored (even by other simulation data projects).

I have a suggestion, please evaluate if it is suitable.

Developers first implement the `calculate` method and demos which can handle non-big data situations (this can be completed this week and next week), then we use new data interfaces to make performance optimization for large-scale data situation.

Maybe the same approach can be used for the our models (Synthesizer).

**Comment by Wh1isper** [comment]
> Developers first implement the calculate method and demos which can handle non-big data situations (this can be completed this week and next week)

It's ok, I'm glad to see that there are practices that can be referenced, it's much better than imagining things out of thin air.

**Comment by MooooCat** [comment]
At present, this PR contains:

- three base classes of synthetic data metric;
- one implementation of column metric;
- one example of column metric.

I prefer to merge this PR at this stage.

**Review by Wh1isper — APPROVED** [review]

**Comment by Wh1isper** [comment]
@all-contributors please add @sjh120  for code

**Pull request merged**